pkg/apis/core/validation/shoot.go

```diff
@@ -878,6 +878,20 @@ func ValidateClusterAutoscaler(autoScaler core.ClusterAutoscaler, fldPath *field
 		allErrs = append(allErrs, field.NotSupported(fldPath.Child("expander"), *expander, availableClusterAutoscalerExpanderModes.List()))
 	}
 
+	if ignoreTaints := autoScaler.IgnoreTaints; ignoreTaints != nil {
+
+		supportedVersion := semver.MustParse("1.16")
+		k8sVersion := semver.MustParse(k8sVersion)
+
+		if k8sVersion.Compare(supportedVersion) >= 0 {
+			if !isIgnoreTaintsUnique(ignoreTaints) {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("ignoreTaints"), ignoreTaints, "must have unique keys"))
```

Comment:
Typically, we provide better information which key is conflicting, see e.g. https://github.com/gardener/gardener/blob/fdc0953d1eb2e607e5ea1ca43fcf0fad539f844b/pkg/apis/core/validation/cloudprofile.go#L176.

Here is some coding how other string slices are checked for duplicates which you might want to get inspired from: https://github.com/gardener/gardener/blob/fdc0953d1eb2e607e5ea1ca43fcf0fad539f844b/pkg/apis/core/validation/shoot.go#L905-L916